### PR TITLE
tweak sort

### PR DESCRIPTION
**Pull request by Lampese**
cc @zxch3n 

## Summary by CodeRabbit

- **Refactor**
	- Enhanced and optimized sorting algorithms to improve performance and maintain code readability.

### CODE REVIEW

File: array/sort_by.mbt [hunk 1]
```diff
@@ -129,21 +124,18 @@ fn try_bubble_sort_by[T](
...
-    for j = i; j > 0 && cmp(arr[j - 1], arr[j]) > 0; j = j - 1 {
-      sorted = false
-      arr.swap(j, j - 1)
-    }
-    if not(sorted) {
-      tries += 1
-      if tries > max_tries {
-        return false
+  let mut sorted = false
+  for t = 1; t < max_tries && not(sorted); t = t + 1 {
```
Comment (zxch3n) [inline array/sort_by.mbt:128]: This modification is wrong. We should avoid sorting the entire `arr` using bubble sort. The original version will abort the bubble sort if it meets `max_tries` unsorted items.

File: array/sort.mbt [hunk 2]
```diff
@@ -291,21 +277,18 @@ fn try_bubble_sort[T : Compare](
...
-    for j = i; j > 0 && arr[j - 1] > arr[j]; j = j - 1 {
-      sorted = false
-      arr.swap(j, j - 1)
-    }
-    if not(sorted) {
-      tries += 1
-      if tries > max_tries {
-        return false
+  let mut sorted = false
+  for t = 1; t < max_tries && not(sorted); t = t + 1 {
```
Comment (zxch3n) [inline array/sort.mbt:281]: This change is probably wrong as well

File: array/sort.mbt [hunk 3]
```diff
@@ -96,16 +96,15 @@ fn merge[T : Compare](arr : ArraySlice[T], mid : Int) -> Unit {
...
-    for p1 = mid - 1, p2 = buf_len - 1, p = mid + buf_len - 1
-        p1 >= 0 && p2 >= 0; {
-      if arr[p1] > buf[p2] {
-        arr[p] = arr[p1]
-        continue p1 - 1, p2, p - 1
-      } else {
-        arr[p] = buf[p2]
-        continue p1, p2 - 1, p - 1
-      }
+  let buf_remaining = for p1 = mid - 1, p2 = buf_len - 1, p = mid + buf_len - 1; p1 >=
```
Comment (zxch3n) [inline array/sort.mbt:99]: This formatting looks kinda weird

**Review by zxch3n — COMMENTED** [review]
Is there a performance change compared to the original version? I wonder which part of the change contributes to that

File: array/sort.mbt (hunk 3 above)
Comment (Yu-zh) [reply array/sort.mbt:99]: Yeah, it's a known issue. We are working on it

**Comment by Lampese** [comment]
@zxch3n I need to apologize for the inappropriate changes, there are parts of this that need to continue to be changed.

**Comment by zxch3n** [comment]
np :)

**Pull request closed without merging**